### ISSUE TITLE
App freezes when config exists on MC.

### ISSUE THREAD

**Issue by Haker120** [sahlberg/libsmb2#214]
As in topic, application NETFS.ELF freezes on 'network initialized' when it detects config (MC0:/SYS-CONF/LAUNCHELF.CNF, not SMB2.cnf, I'm unable to test other server), when I saved from this particular version of wLE. I tested on SCPH-50004, and I am confused and amazed how fast it transfered my VCD games! 

Can it load settings from pfs0:/__sysconf/FMCB/LAUNCHELF.CNF? I guess not as it doesn't detect this cnf from 'vanila' version of wLE.

**Comment by sahlberg** [comment]
On Tue, Oct 12, 2021 at 1:31 AM Haker120 ***@***.***> wrote:
>
> As in topic, application freezes on 'network initializing' when it detects config (LAUNCHELF.CNF, not SMB2.cnf, I'm unable to test other server), when I saved from this particular version of wLE. I tested on SCPH-50004, and I am confused and amazed how fast it transfered my VCD games!
>
> Can it load settings from pfs0:/__sysconf/FMCB/LAUNCHELF.CNF? I guess not as it doesn't detect this cnf from 'vanila' version of wLE.

Nope, I havent added that yet.
I can try to add that in a week or two. My build system has crached so
I need to re-install and set up PS2DEV
but PS2DEV is not exactly easy to set up and get working on Fedora :-(

**Comment by Haker120** [comment]
Alright, good luck restoring it also main issue persists, and you haven't commented on this, LAUNCHELF.CNF freezes it after network initialized, Other than this continue your great work! :)

**Comment by sahlberg** [comment]
On Fri, Oct 15, 2021 at 10:41 PM Haker120 ***@***.***> wrote:

> Alright, good luck restoring it also main issue persists, and you haven't
> commented on this, LAUNCHELF.CNF freezes it after network initialized,
> Other than this continue your great work! :)
>
Got the build environment working.
Try this:  https://github.com/sahlberg/wLaunchELF/blob/master/NETFS.ELF
It fixes a bug where it might crash/hang if it sees the same entry in the
config twice.


For LAUNCHELF.CNF,  I don't know what that file does and don't think I did
anything that should affect it.
What does that file do and how do you use it?

**Comment by sahlberg** [comment]
On Sun, Oct 17, 2021 at 7:46 PM Haker120 ***@***.***> wrote:

> Launchelf.CNF is a file that stores configuration for u/wLE like shorts in
> start menu in it where you see O to file manager, triangle to About wLE
> etc, as for now your fork sees two locations, MC?:/SYS-CONF and
> USB:/SYS-CONF/
>
> I'm going to test version you provided and I'll update my comment. ;)
>

Please try it.
Also, I know almost nothing about ps2 or development or tools. I just
barely know enough to get my libraries to compile and work with wLe.
If I ask a stupid question, it is just because I know very little/nothing
about ps2 programmeing and tools :-)

**Comment by Haker120** [comment]
I tested, it has a problem with marks but loads settings from MC, and managed to corrupt MBR but SP193's app ps2 HDD checker helped, but hanged at 10% but fixed issue with not loading FHDB after using this version of wLE, I'm not sure what caused this issue tho.

And it's not a problem as long as it works, I agreed to test it so any corruptions are part of it. Shame SMB2/3 cannot be added to official version or even OPL.

I don't even know how to properly use github and all I did is to open issues when I find any problems with my fav apps. xD
